File: c/estimate_pi.c
#include<stdio.h>
#include<math.h>

#define LOWER 1e-15

double fact (double);
double count(double);
double pow  (double,double);

int main(void)
{
	double k = 1.0;
	double item , summed;
	double pi;

	item = count (k);
	summed = item;
	printf("Enter loop:summed = %f.\n",summed);
	while (item >= LOWER)
	{
		++k;
		item = count(k);
		summed += item;
	}
//	printf("Exit loop:summed = %f.\n",summed);
	pi = sqrt(2)*2.0 / 9801;
	pi *= summed;
	pi = 1.0 / pi;
	printf("%f\n",pi);
	return 0;
}
double fact(double k)
{
	if (k > 0)
	{
		return k  * fact(k-1.0);
	}
	else
	{
		return 1.0;
	}
}
double count(double k)
{
	return	(fact(4.0*k)*(1103.0 + 26390.0* k)) 	/ 
		(pow(fact(k),4.0)*pow(396.0,4.0*k))	;
}
double pow(double x,double y)
{
	double n = 1;
	while (y > 0)
	{
		y--;
		n *= x;
	}
	return n;
}

File: python/estimate_pi.py
#estimate_pi.py

def fact (k):
	if k > 0:
		return k * fact (k - 1)
	else:
		return 1
def count(k):
	up = fact(4*k)*(1103 + 26390 * k)
	down = fact(k)**4*396**(4*k)
	up = up * 1.0
	down = down * 1.0
	return up / down
k = 0
item = count(k)
summed = 0 + item
#print (summed)
while item >= 1e-15:
	k = k + 1
	item = count(k)
	summed = summed + item
#print (summed)
import math
pi = (2 * math.sqrt(2)) / 9801
pi = pi * summed
pi = 1.0 / pi
print("pi = ",end="")
print (pi)
print("math.pi = ",end="")
print (math.pi)
#print (abs(pi - math.pi)) 
